cli/mk_data/src/generate.rs

```diff
@@ -225,24 +225,30 @@ pub fn generate_category_jobs<'a>(
 ) -> Result<Vec<Job>, Error> {
     let mut jobs: Vec<Job> = vec![];
     for (name, raw_spec) in raw_specs {
-        let filename = output_dir.join(category).join(format!("{}.yaml", name));
-        if !force && filename.exists() {
-            continue;
-        }
-        if !force {
-            if let Some(explicit_check_path) = &raw_spec.skip_if_output_exists {
-                let filename = output_dir.join(explicit_check_path);
-                if filename.exists() {
+        let response_filename = output_dir.join(category).join(format!("{}.yaml", name));
+        match (force, raw_spec.skip_if_output_exists.as_ref()) {
+            (false, Some(path)) => {
+                let check_path = output_dir.join(path);
+                if check_path.exists() {
+                    trace!(name, explicit = true, path = %check_path.display(), "skipping job because output exists");
                     continue;
                 }
-            }
+            },
+            (false, None) => {
+                if response_filename.exists() {
+                    trace!(name, explicit = false, path = %response_filename.display(), "skiping job because output exists");
+                    continue;
+                }
+            },
+            (true, _) => {},
```

Comment:
I think we should recreate if either of `response_filename` or `skip_if_output_exists` don't exist because it's confusing that sometimes you need to delete the response file and sometimes you need to delete the directory, which may or may not contain the response file.

Sidenote: we also have a mixture of the response file being either within or parallel to `generated_data/envs/<thing>/` and there's a lot of boilerplate to move everything into the right place each `config.toml` definition. Standardising that in the future might make this easier.

Reply:
I explicitly chose to do it this way so that if you set `skip_if_output_exists` it can override what's checked for regeneration. When this job runs it will overwrite the response file anyway.

Re: sidenote: I've noticed that as well. If we get a chance to do greenkeeping this week, I want to work on https://github.com/flox/flox/issues/3081 to address that and a variety of other things.